### ISSUE TITLE
[BUG] crc start and stop operations are very slow

### ISSUE THREAD

**Issue by tofflos** [code-ready/crc#617]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```Powershell
PS C:\Users\Erik> crc version
version: 1.0.0-beta.5+f2aa58c
OpenShift version: 4.1.14 (embedded in binary)
```
  
## CRC status
```Powershell
PS C:\Users\Erik> crc status
CRC VM:          Running
OpenShift:       Running (v4.x)
Disk Usage:      11.12GB of 16.09GB (Inside the CRC VM)
Cache Usage:     11.23GB
Cache Directory: C:\Users\Erik\.crc\cache
```

## CRC config
```bash
# Put `crc config view` output here
```

## Host Operating System
```Powershell
PS C:\Users\Erik> systeminfo

Host Name:                 MSI
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.18362 N/A Build 18362
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Windows-användare
Registered Organization:
Product ID:                00330-80000-00000-AA556
Original Install Date:     2019-06-25, 21:01:37
System Boot Time:          2019-09-19, 19:45:03
System Manufacturer:       MICRO-STAR INTERNATIONAL CO., LTD
System Model:              MS-ACB311
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 60 Stepping 3 GenuineIntel ~2801 Mhz
BIOS Version:              American Megatrends Inc. EACB3IMS V8.1, 2014-07-10
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume4
System Locale:             sv;Svenska
Input Locale:              sv;Svenska
Time Zone:                 (UTC+01:00) Amsterdam, Berlin, Bern, Rom, Stockholm, Wien
Total Physical Memory:     16 306 MB
Available Physical Memory: 746 MB
Virtual Memory: Max Size:  23 084 MB
Virtual Memory: Available: 2 909 MB
Virtual Memory: In Use:    20 175 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\MSI
Hotfix(s):                 8 Hotfix(s) Installed.
                           [01]: KB4514359
                           [02]: KB4498523
                           [03]: KB4503308
                           [04]: KB4508433
                           [05]: KB4509096
                           [06]: KB4515383
                           [07]: KB4516115
                           [08]: KB4515384
Network Card(s):           4 NIC(s) Installed.
                           [01]: Killer E2200 Gigabit Ethernet Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [02]: Intel(R) Dual Band Wireless-AC 3160
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.0.1
                                 IP address(es)
                                 [01]: 192.168.0.107
                                 [02]: fe80::50f3:1c42:65dd:dce5
                           [03]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth-nätverksanslutning 2
                                 Status:          Media disconnected
                           [04]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.17.197.129
                                 [02]: fe80::4c2d:6160:f2bc:d8ac
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. 
  2. 
  3. 
  4. 

### Expected
It takes less than a minute to start and less than ten seconds to stop.

### Actual
It takes close to ten minutes to start and close to two minutes to stop.

### Logs

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

```Powershell
PS C:\Users\Erik> Measure-Command { crc setup | Out-Host } | Select-Object -Property TotalSeconds
INFO Checking if running as normal user
INFO Caching oc binary
INFO Unpacking bundle from the CRC binary
INFO Check Windows 10 release
INFO Hyper-V installed
INFO Is user a member of the Hyper-V Administrators group
INFO Does the Hyper-V virtual switch exist
Setup is complete, you can now run 'crc start' to start a CodeReady Containers instance

TotalSeconds
------------
   2,6055306
```

```Powershell
PS C:\Users\Erik> Measure-Command { crc start --pull-secret-file .\Downloads\pull-secret | Out-Host } | Select-Object -Property TotalSeconds
INFO Checking if running as normal user
INFO Checking if oc binary is cached
INFO Check Windows 10 release
INFO Hyper-V installed and operational
INFO Is user a member of the Hyper-V Administrators group
INFO Does the Hyper-V virtual switch exist
INFO Loading bundle: crc_hyperv_4.1.14.crcbundle ...
INFO Creating VM ...
INFO Verifying validity of the cluster certificates ...
INFO Will run as admin: add dns server address to interface vEthernet (Default Switch)
INFO Check internal and public dns query ...
WARN Failed Public dns query: ssh command error:
command : host -R 3 quay.io
err     : Process exited with status 1
output  : ;; connection timed out; no servers could be reached
 :
INFO Copying kubeconfig file to instance dir ...
INFO Adding user's pull secret and cluster ID ...
INFO Starting OpenShift cluster ... [waiting 3m]
INFO
INFO To access the cluster, first set up your environment by following 'crc oc-env' instructions
INFO Then you can access it by running 'oc login -u developer -p developer https://api.crc.testing:6443'
INFO To login as an admin, username is 'kubeadmin' and password is XXXXX-XXXXX-XXXXX-XXXXX
INFO
INFO These credentials can also be used to access the OpenShift web console at https://console-openshift-console.apps-crc.testing
CodeReady Containers instance is running

TotalSeconds
------------
 515,5249251
```

```Powershell
PS C:\Users\Erik> Measure-Command { crc stop | Out-Host } | Select-Object -Property TotalSeconds
Stopping CodeReady Containers instance... this may take a few minutes
CodeReady Containers instance stopped

TotalSeconds
------------
 103,9279981
```
```Powershell
PS C:\Users\Erik> Measure-Command { crc delete | Out-Host } | Select-Object -Property TotalSeconds
CodeReady Containers instance deleted

TotalSeconds
------------
   1,7423171
```

**Comment by gbraad** [comment]
> It takes close to ten minutes to start and close to two minutes to stop.

This is not an issue, but known behaviour. You start a full (production-like) cluster on your host, with almost all operators enabled as like on a cloud deployment. The sttart copies the disk image, which dependent on your machine can take a few seconds to several minutes (9GB of data). after which the machine gets started (about 10secs to 2 minutes depending on your performance) and then it needs to change/inject the pull-secret (about 1 to 2 mins) followed by actually starting the cluster abd operators (about 3 mins or more). This is really dependent on the machines performance. I can start the cluster within less than 5 minutes, while on a laptop (i7, 32GB it can take longer as it does not use an SSD)

The same for stop; we are sendin ga graceful shutdown (ACPI call), which causes the cluster to shutdown, and again this depends on your performance. It takes less than 20 secs on my machine, but when actively used (many containers in running state) it could take longer.

OpenShift will in later iterations look at optimizing some of these points and so will we, but at the moment this is not the highest priority.

Will close and mark this as invalid (known) issue.

**Comment by neodingo** [comment]
I have a fresh windows 10 install on a Ryzen 3600, 32gb ram, and a 1tb samsung nvme drive on an ASRock Phantom Gaming 4 and it has been a few hours at this state:

```
PS C:\> .\crc.exe setup
INFO Checking if oc binary is cached
INFO Checking if podman remote binary is cached
INFO Checking if goodhosts binary is cached
INFO Checking if CRC bundle is cached in '$HOME/.crc'
INFO Checking minimum RAM requirements                                                                                                                                       INFO Checking if running as normal user                                                                                                                                      INFO Checking Windows 10 release                                                                                                                                             INFO Checking Windows edition                                                                                                                                                INFO Checking if Hyper-V is installed and operational                                                                                                                        INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
Setup is complete, you can now run 'crc start' to start the OpenShift cluster
PS C:\> .\crc.exe start --log-level debug
DEBU CodeReady Containers version: 1.16.0+004e857
DEBU OpenShift version: 4.5.0-0.okd-2020-10-15-235428 (embedded in binary)
DEBU Total memory of system is 34290573312 bytes
WARN A new version (1.18.0) has been published on https://cloud.redhat.com/openshift/install/crc/installer-provisioned
INFO Checking if oc binary is cached
DEBU Running 'C:\Users\michael\.crc\bin\oc\oc.exe version --client'
DEBU Found oc.exe version 4.5.0-0.okd-2020-10-15-235428
DEBU oc binary already cached
INFO Checking if podman remote binary is cached
DEBU Currently podman remote is not supported
INFO Checking if goodhosts binary is cached
DEBU goodhost binary already cached
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34290573312 bytes
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking Windows edition
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
? Image pull secret [? for help] *******************************************

INFO Loading bundle: crc_hyperv_4.5.0-0.okd-2020-10-15-235428.crcbundle ...
INFO Checking size of the disk image C:\Users\michael\.crc\cache\crc_hyperv_4.5.0-0.okd-2020-10-15-235428\crc.vhdx ...
INFO Creating CodeReady Containers VM for OpenShift 4.5.0-0.okd-2020-10-15-235428...
```

No further logs and no progress thus far, is this possibly something different than expectation management?

**Comment by gbraad** [comment]
Please file a new issue as you need to provide the requested info. Also note that this issue has been closed for over a year.
